lab_share_lib/processing/rabbit_message_processor.py

```diff
@@ -72,6 +72,12 @@ def process_message(self, headers, body):
         except TransientRabbitError as ex:
             LOGGER.error(f"Transient error while processing message: {ex.message}")
             raise  # Cause the consumer to restart and try this message again.
+        except KeyError as ex:
+            LOGGER.error(
+                f"Unrecoverable error: the required subject {ex.args[0] if len(ex.args) > 0 else ''} "
+                f"is not found in the defined message_subjects dictionary in configs."
+            )
+            return False
```

Comment:
I meant we should create a new try block above this one so we're not expecting this one try block to do two separate operations.

Reply:
Didn't quite get what you meant. Are you suggesting a nested try block?

Reply:
Or something like:

![image](https://github.com/user-attachments/assets/1906a555-1fbd-4fae-bc8c-78918a5fb0cf)

Reply:
Yes the second one.  The two in the original `try` block were only generating exceptions specific to themselves and so putting them in a single `try`/`catch` section of code makes it harder to understand why might throw which exception.  Doing what you're showing in the screenshot may be more code, but it makes it more maintainable and easier to understand.